### ISSUE TITLE
Evaluating a saved checkpoint requires constructing a full training rig

### ISSUE THREAD

**Issue by Eric-Louchez** [Shorokoo/Shorokoo#329]
There is no route from a saved checkpoint to a runnable model that does not go through
`TrainingRig`. Every loading entry point either is a rig method or needs one:

* `rig.LoadCheckpoint(path)` / `rig.LoadCheckpointFromSkpt(path)` — instance methods, so a rig must
  exist first, and `training.md` requires it to be "built from the **same** model/loss/optimizer
  graphs".
* `TrainingRig.Load(path)` — rebuilds the rig from the file; still a rig.
* `checkpoint.ToInferenceModel()` — the thing actually wanted — throws without one (verified on `main`
  798953d): *"ToInferenceModel() requires a training rig, but this checkpoint has none attached."*
* `Persistence.LoadTrainingCheckpoint(path, trainableDef, modelStateDef, optimizerStateDef)` takes
  struct defs rather than a rig, but those defs are themselves rig properties.

So computing a validation loss means composing the loss graph, running autodiff, lowering the
optimizer per parameter and running every initializer — then discarding all of it except the weights,
which are immediately overwritten from disk.

## Measured

A forward-only evaluation of a 49,214,208-parameter decoder-only transformer over 512 validation
batches (2,097,152 tokens), CUDA backend:

```
  ... 65 s of rig build ...
FINAL val loss 10.801927 nats/token over 2,097,152 tokens
{ "eval_seconds": 76.03 }
```

**76 seconds, of which 65 is building a training rig for a job that computes no gradients** — and
roughly 34 of those 65 are `Initialize: InitializeModelParams` generating 49.2 M random floats that
`LoadCheckpoint` overwrites on the next line. Across six planned evaluation processes that is ~8
minutes of pure waste, on top of the same cost paid by every training process and every resume
(#327).

## It also distorted the model's design

The loss was put *inside* the model — the model takes `(tokens, targets)` and returns the scalar
cross-entropy, with a pass-through module in the rig's loss slot (#331) — partly so that
`ToInferenceModel()` would hand back something that computes a validation loss directly. The
alternative, authoring a second forward-only model and binding the trained weights into it by name,
runs into parameter naming (#322): a separately-authored graph has no reliable way to receive a
training checkpoint's weights. Given that naming, the rig is not just the documented path to a
checkpoint's contents; it is the only trustworthy one.

Related: #121 (a state-only `.skpt` container / weightless entry), #327 (what the build costs),
#322 (naming), #331 (the pass-through loss slot).

**No migration work.** Shorokoo has no installed base to protect. Add a rig-free load path and change
the persisted layout outright — no compatibility shim, no version bump, no format-version
negotiation, and no migration path for checkpoints written by an earlier build.
